### DISCUSSION TITLE
Isomorphic use case...How does alasql performance compare with server side libraries?

### DISCUSSION THREAD

**Discussion by btakita** [agershun/alasql#1876] [Q&A] [answered]
I'm considering using AlaSQL on the server side for isomorphic libraries. How does the performance of `alasql` on node.js compare to `better-sqlite3`? On Bun, how does it's performance compare to `bun:sqlite`?

Can alasql load to the sqlite db on the file system & save to it automatically?

**Comment by mathiasrw** [comment]
If you _can_ use sqlite then use sqlite.

**Reply by paulrutter** [reply, marked as answer]
We thoroughly compared better-sqllite3 against alasql, but went for alasql n the end because the data ingestion overhead of better-sqllite3 was too large when compared to alasql (which doesn't have this overhead since data can be queried in the node runtime directly).

For query performance, your mileage will vary depending on the query complexity. Joins over more than one table, with lots of rows, will be faster in sqlite for example.

**Reply by mathiasrw** [reply]
I am happy to hear that we are providing a useful alternative. Feel free to share some details from your performance test so we can "brag" a bit about it :)
